ui/src/dashboards/actions/index.ts

```diff
@@ -212,6 +213,65 @@ export const removeDashboardLabels = (
 
 // Thunks
 
+export const createDashboard = () => async (
+  dispatch,
+  getState: GetState
+): Promise<void> => {
+  try {
+    const {
+      orgs: {org},
+    } = getState()
+
+    const newDashboard = {
+      name: DEFAULT_DASHBOARD_NAME,
+      cells: [],
+      orgID: org.id,
+    }
+
+    const data = await createDashboardAJAX(newDashboard)
+    dispatch(push(`/orgs/${org.id}/dashboards/${data.id}`))
+    dispatch(checkDashboardLimits())
+  } catch (error) {
+    console.error(error)
+
+    if (isLimitError(error)) {
+      const message = extractMessage(error)
+      dispatch(notify(copy.resourceLimitReached('dashboards', message)))
+    } else {
+      dispatch(notify(copy.dashboardCreateFailed()))
+    }
+  }
+}
+
+export const cloneDashboard = (dashboard: Dashboard) => async (
+  dispatch,
+  getState: GetState
+): Promise<void> => {
+  try {
+    const {
+      orgs: {org},
+      dashboards,
+    } = getState()
+
+    const allDashboardNames = dashboards.list.map(d => d.name)
+
+    const clonedName = incrementCloneName(allDashboardNames, dashboard.name)
+
+    const data = await client.dashboards.clone(dashboard.id, clonedName, org.id)
+
+    dispatch(push(`/orgs/${org.id}/dashboards/${data.id}`))
```

Comment:
the router change should be last thing we do in a sequence, if the action after this is needed for the page then doing this here hides the implicit coupling and makes it harder to debug in the future.

Reply:
👍